packages/toolpad-components/src/DataGrid.tsx

```diff
@@ -315,6 +342,11 @@ const DataGridComponent = React.forwardRef(function DataGridComponent(
     [getRowId, columns],
   );
 
+  const hasImageColumns = React.useMemo(
+    () => !!columns.find((column) => column.type === 'image'),
```

Comment:
if you don't need a value that is found, you could also use [Array.some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) method in addition could slightly shorten the expression:
```suggestion
    () => columns.some(({ type }) => type === 'image'),
```